acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/web3j/privacy/PrivacyClusterAcceptanceTest.java

```diff
@@ -215,7 +228,123 @@ public void node2ExpectError() {
             .keyPair(keypair2)
             .build(TransactionType.STORE);
 
-    node2.execute(transactions.createPrivateRawTransaction(invalidStoreValueFromNode2));
+    node2.execute(privateTransactions.createPrivateRawTransaction(invalidStoreValueFromNode2));
+  }
+
+  @Test
+  public void node1CanInteractWithMultiplePrivacyGroups() {
+    final String ABC =
+        "0x393579496e2f4f59545a31784e3753694258314d64424a763942716b364f713766792b37585361496e79593d";
+    final String AB =
+        "0x4479414f69462f796e70632b4a586132594147423062436974536c4f4d4e6d2b53686d422f374d364334773d";
+    final String contractForABC = "0x1efee0ab2c1ec40c4b48410e5832d254c2eda0b0";
+
+    long nextNonce = getNonce(node1, ABC);
+
+    final String deployContractForABC =
+        PrivateTransactionBuilder.builder()
+            .nonce(nextNonce)
+            .from(node1.getAddress())
+            .to(null)
+            .privateFrom(BytesValue.wrap(PUBLIC_KEY_1.getBytes(UTF_8)))
+            .privateFor(
+                Lists.newArrayList(
+                    BytesValue.wrap(PUBLIC_KEY_2.getBytes(UTF_8)),
+                    BytesValue.wrap(PUBLIC_KEY_3.getBytes(UTF_8))))
+            .keyPair(keypair1)
+            .build(TransactionType.CREATE_CONTRACT);
+
+    String transactionHash =
+        node1.execute(privateTransactions.deployPrivateSmartContract(deployContractForABC));
+
+    privateTransactionVerifier
+        .validPrivateContractDeployed(contractForABC)
+        .verify(node1, transactionHash);
+
+    nextNonce = getNonce(node1, ABC);
+
+    final String storeValueForABC_first =
+        PrivateTransactionBuilder.builder()
+            .nonce(nextNonce)
+            .from(node1.getAddress())
+            .to(Address.fromHexString(contractForABC))
+            .privateFrom(BytesValue.wrap(PUBLIC_KEY_1.getBytes(UTF_8)))
+            .privateFor(
+                Lists.newArrayList(
+                    BytesValue.wrap(PUBLIC_KEY_2.getBytes(UTF_8)),
+                    BytesValue.wrap(PUBLIC_KEY_3.getBytes(UTF_8))))
+            .keyPair(keypair1)
+            .build(TransactionType.STORE);
+
+    transactionHash =
+        node1.execute(privateTransactions.createPrivateRawTransaction(storeValueForABC_first));
+
+    privateTransactionVerifier.validEventReturned("1000").verify(node1, transactionHash);
```

Comment:
Also, does refer to the "03e8" (hex for 1000) at the end of PrivateTransactionBuilder.SET_FUNCTION_CALL?   another reason that generating the literals instead of storing already assembled hex strings would be beneficial.  But that's for another issue.

Reply:
agreed